File: PopOpen/ProcessFinder.cs
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;

namespace PopOpen
{
    public class ProcessFinder : IFindProcess
    {
        public Process Find(string path)
        {
            return InUseDetection.GetProcessesUsingFiles(new List<string> {path}).First();
        }
    }
}

File: Pop/ProcessFinder.fs
namespace Pop

open System.Diagnostics

module ProcessFinder =
    let Find (filePath: string) =
        let filePaths = [ filePath ]

        let processes = 
            filePaths
            |> List.toSeq
            |> Cs.InUseDetection.GetProcessesUsingFiles
            |> List.ofSeq<Process>

        match processes.Length with
        | 0 -> nativeint 0
        | _ -> processes.Head.MainWindowHandle

        
                



